src/DataProtection/DataProtection/src/KeyManagement/XmlKeyManager.cs

```diff
@@ -477,11 +477,11 @@ private IAuthenticatedEncryptorDescriptorDeserializer CreateDeserializer(string
         {
             return _activator.CreateInstance<AuthenticatedEncryptorDescriptorDeserializer>(descriptorDeserializerTypeName);
         }
-        else if (type == typeof(CngCbcAuthenticatedEncryptorDescriptorDeserializer))
+        else if (type == typeof(CngCbcAuthenticatedEncryptorDescriptorDeserializer) && RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
         {
             return _activator.CreateInstance<CngCbcAuthenticatedEncryptorDescriptorDeserializer>(descriptorDeserializerTypeName);
         }
-        else if (type == typeof(CngGcmAuthenticatedEncryptorDescriptorDeserializer))
+        else if (type == typeof(CngGcmAuthenticatedEncryptorDescriptorDeserializer) && RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
Is this netstandard 2.0? These APIs aren't trim friendly.

Reply:
Yes, it targets netstandard2.0.

What specifically isn't trim-friendly? No warnings are coming from the linker.

Reply:
My bad, I meant that `RuntimeInformation.IsOSPlatform(OSPlatform.Windows)` isn't recognized by the trimmer. Those branches will exist when this is trimmed and we're targeting linux.

Reply:
Got it. I added `IsOSPlatform` to suppress errors from build, not for trimming.

A couple of types won't matter. I don't think introducing a net6-window target is worth it.